sycl/include/sycl/ext/oneapi/device_global/device_global.hpp

```diff
@@ -63,14 +68,49 @@ class device_global_base {
   pointer_t get_ptr() noexcept { return usmptr; }
   pointer_t get_ptr() const noexcept { return usmptr; }
 
+  template <typename, typename, typename> friend class device_global_base;
+
+#ifndef __SYCL_DEVICE_ONLY__
+  template <typename OtherT, typename OtherProps>
+  static constexpr const T &
+  ExtractInitialVal(const device_global_base<OtherT, OtherProps> &Other) {
+    if constexpr (OtherProps::template has_property<device_image_scope_key>())
+      return Other.val;
```

Comment:
The object being constructed cannot have the `device_image_scope` property, but `other` can. Otherwise, we could have just lined the `Other.init_val` in the ctor.